src/modules/git_status.rs

```diff
@@ -52,6 +54,12 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
                 "style" => Some(Ok(config.style)),
                 _ => None,
             })
+            .map(|variable| match variable {
+                "lfs" => context
+                    .exec_cmd("git", &["config", "--get", "lfs.repositoryformatversion"])
+                    .map(|_| Ok(config.lfs)),
```

Comment:
Please try handling this with `git2` or `git-config` for better performance. Any git operations should also pass along the repo paths, as in some instances the current directory might not match the starship working directory.

Reply:
given that starship runs at the time a prompt is generated when would the directories not match?

Reply:
It's true that the real-runtime conditions for this are not that common. It's possible for there to be a mismatch between cwd and the passed directory due to shenanigans with logical paths and symlinks, or because someone is parsing the output of starship as a way to get information about directories (yes, we have seen at least one real instance of this).

However, in my personal opinion, the bigger issue here is just modularity and testability of the code. Historically, many pieces of starship were tied to the environment, and once we started hitting bugs, we had to extract them into their own pieces so that we could mock those tests. The most significant example of this is `get_env()` in our context manager, but there are lots of other small examples throughout the codebase.

Reply:
@chipbuster do you think it would be better broken out into a separate module for git extensions?

Reply:
I don't have a great feel for what the best architecture for this is.

On one hand, it would be nice to have more modularity so that we can add other information (e.g. #3644) without having to stuff a million options into each git module. On the other hand, modules cannot communicate with each other (outside of caching information for each other in the OnceCells), so there's risk of duplicate work, as well as the problem of just having 10 different git modules that each do different things.

Can you think of other git extensions that might be added at some point?

Reply:
Re potential cwd issues: PowerShell will not set the cwd location in the prompt function. Though considering it again, `context.exec_cmd` will properly handle this. But anyway, the module has all the git directory locations at hand, so it should use that information.

Reply:
The most obvious one would be a heuristic (based on looking at branch names) to remind the user in the prompt that `git flow` is in use in a repo.

It is unfortunately hard to search for git extensions, because there's a Windows thing called "git extensions" which appears to not be a git extension. Argh!